traitsui/group.py

```diff
@@ -159,16 +159,27 @@ class Group(ViewSubElement):
     #: the group and its items are not visible (and they disappear if they were
     #: previously visible). If the value evaluates to True, the group and items
     #: become visible. All **visible_when** conditions are checked each time
-    #: that any trait value is edited in the display. Therefore, you can use
-    #: **visible_when** conditions to hide or show groups in response to user
-    #: input.
+    #: that any trait value on an object in the UI's context is changed.
+    #: Therefore, you can use **visible_when** conditions to hide or show
+    #: groups in response to user input. Be aware that this only applies to
+    #: traits in the UI's context. As a result, changes to nested traits may
+    #: not trigger the expression to be checked they don't also change a trait
+    #: on some object in the context. Additionally, the expression needs to be
```

Comment:
I reordered the sentence a little and fixed what I think is a typo - `they` to `that` in `... they dont also change a trait ...`

If you are happy with this rewording, these changes need to be propogated to the other docstrings.

```suggestion
    #: traits in the UI's context. As a result, changes to nested traits that
    #: don't also change a trait on some object in the context may not
    #: trigger the expression to be checked. Additionally, the expression
    #: needs to be
```